library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php

```diff
@@ -76,6 +77,19 @@ public function execute(CompanyInterface $company)
         $domain = $this->entityTools
             ->persistDto($domainDto, $domain);
 
-        $company->setDomain($domain);
+        /** @var CompanyDto $companyDto */
+        $companyDto = $this
+            ->entityTools
+            ->entityToDto($company);
+
+        $companyDto
+            ->setDomain($domainDto);
+
+        $this
+            ->entityTools
+            ->updateEntityByDto(
+                $company,
+                $companyDto
+            );
```

Comment:
The previous service (library/Ivoz/Provider/Domain/Service/Domain/UpdateByBrand.php ) uses persistDto while this uses updateEntityByDto. Is POST/PRE persist the reason behind this?

Reply:
yes it is